Fix responsive spacing issues for show map button

kccrs opened the pull request:
# Purpose 

On smaller screen sizes the show map button was resizing too small and did not have enough breathing room between its neighbor.

I also updated many elements including the header as it had some responsive issues of its own. There were many challenges in that many of the elements have the same classname and were styled the same even though the layout was different between the admin and user pages.  I added additional classnames to some of the elements in order to ensure layout differences remained intact. 

I had to manually copy some code from master into my branch that didn't come over when I pulled down for some reason, so that is why some of the changes appear in Admin and App components.  I did not want to lose the already approved and merged changes.

swanie21 commented on lib/components/App.js:
```diff
@@ -184,46 +183,37 @@ export default class App extends Component {
-    }
-    firebase.database().ref(`speakers/${speaker.key}`).update( speaker );
-  }
-
```
Do we not need this function anymore for gear ups or the empty array as the state for gearUp? I still have it in my master.

Jeff-Duke replied:
We definitely need this still.  This is essential for the RSVP functionality on the guest speaker.

kccrs replied:
Hmm... for some reason when I pulled down master, it didn't bring all the changes with it.... I copied and pasted stuff that I thought was missing, but I guess I missed a few things.

kccrs replied:
Fixed.

swanie21 commented on lib/components/Header.js:
```diff
@@ -5,7 +5,7 @@ import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
 const Header = ({ admins, user, createSpike, showForm, toggleForm }) => {
   return (
-    <header className="Header">
+    <header>
```
Good catch to remove the redundancy.

kccrs replied:
yeah we weren't using it in our Sass. Since there's only one header, we didn't need a class.  :)

swanie21 commented on lib/components/App.js:
```diff
@@ -236,29 +226,18 @@ export default class App extends Component {
             updateCount={(spike) => this.updateCount(spike)}
-            assignRooms={this.assignRooms}
-            admins={this.state.admins}
-            deleteSpike={(spike) => this.deleteSpike(spike)}
```
Are we not passing a lot of these props to components anymore?

Jeff-Duke replied:
Same as above, I like deconstructing state to pass down props, we need to make sure we're still passing down methods like updateSpeakerAttendees, deleteSpike, updateSpike etc.

kccrs replied:
Fixed.

swanie21 commented:
It looks like the show map button was fixed, which was the main purpose of this PR and a lot of other changes were made. On mobile the button fonts on these buttons: Admins List, Show Map, Add Guest Speaker, Add Gear Up, Sign Out and Add Spike get smaller, which I think it should be the same size as desktop or even larger since it's on a smaller device, so users can see the text easier. Also the sign out button is briefly at the top of the profile pic before it pops down below the pic at a media query. The header responsive design definitely looks better though. I think the sign out button shouldn't be popping around though and with the media query that aligns the profile pic in the middle it should just go beneath it at that point. 

It looks like there was a lot of JS changes on the App component. Can you help explain why these changes were made? 

Otherwise the responsive design looks much better!

Jeff-Duke commented:
This PR breaks the guest speaker RSVP functionality and gets rid of the gear-up section.  We need to make sure we have existing functionality before merging this PR.

Jeff-Duke commented on lib/components/App.js:
```diff
@@ -184,46 +183,37 @@ export default class App extends Component {
-            sanitizeInput={this.sanitizeInput}
+            user={user}
+            speaker={speaker}
           />
```
We definitely still need to pass these functions down.  I see that you're deconstructing state to make things cleaner but we still need the methods.

kccrs replied:
Yep, you're right... I lost those in the merge.  Fixed.

kccrs commented:
Ok, I think I have fixed all the issues in this PR and updated a few other things that I noticed... you know.. change one thing, something else breaks.  Pull it down, take a look and make sure everything looks OK and is working correctly.

Jeff-Duke commented:
As far as I can tell this all looks good now.  Can someone else pull down the branch and confirm?
